src/main/java/Duke.java

```diff
@@ -1,10 +1,292 @@
+                    tmpTask instanceof Deadline? "D" : "E");
+            String isDone = (tmpTask.getIsDone()? "1" : "0");
+
+            if (taskType.equals("T")) {
+                String taskDesc = tmpTask.getTask();
+                strToWrite = String.join(">", taskType, isDone, taskDesc);
+            } else {
+                String posText = "";
+                String addInfo = "";
+
+                if (taskType.equals("D")) {
+                    posText = " (by:";
+                    addInfo = ((Deadline) tmpTask).getBy();
+                } else if (taskType.equals("E")) {
+                    posText = " (at:";
+                    addInfo = ((Event) tmpTask).getAt();
+                }
+                int pos = tmpTask.getTask().indexOf(posText);
+                String taskDesc = tmpTask.getTask().substring(0, pos);
+                strToWrite = String.join(">", taskType, isDone, taskDesc, addInfo);
+            }
+            fw.write(strToWrite + System.lineSeparator());
+        }
+        fw.close();
+    }
+
+    // add task to allTasks
+    public static void addTask(String[] inputArr) {
+
+        if (inputArr.length == 1) {
+            System.out.println(INDENT + "Oops, the description field cannot be empty!");
+            return;
+        }
+
+        Task newTask;
+        
+        switch(inputArr[0]) {
+        case("todo"): {
+            String[] descArr = Arrays.copyOfRange(inputArr, 1, inputArr.length);
+            newTask = new Todo((convertToString(descArr)));
+            break;
+        }
+        case("deadline"): {
+            try {
+                int pos = Arrays.asList(inputArr).indexOf("/by");
+                String[] descArr = Arrays.copyOfRange(inputArr, 1, pos);
+                String[] byArr = Arrays.copyOfRange(inputArr,pos+1,inputArr.length);
+                newTask = new Deadline(convertToString(descArr),convertToString(byArr));
+                break;
+            } catch (IllegalArgumentException e) {
+                System.out.println(INDENT + "Sorry, you did not specify a deadline!");
+                return;
+            }
+        }
+        case("event"): {
+            try {
+                int pos = Arrays.asList(inputArr).indexOf("/at");
+                String[] descArr = Arrays.copyOfRange(inputArr, 1, pos);
+                String[] atArr = Arrays.copyOfRange(inputArr, pos + 1, inputArr.length);
+                newTask = new Event(convertToString(descArr), convertToString(atArr));
+                break;
+            } catch (IllegalArgumentException e) {
+                System.out.println(INDENT + "Sorry, you did not specify a time!");
+                return;
+            }
+        }
+        default: {
+            newTask = new Task((convertToString(inputArr)));
+        }
+        }
+        allTasks.add(newTask);
+        printNewlyAddedTask(newTask);
+    }
+
+    // delete task from allTasks
+    public static void deleteTask(int num) {
```

Comment:
You have added all the different commands as methods in your Duke.java. Please consider refactoring your code to include more OOP.